CONTRIBUTING.md

```diff
@@ -1,18 +1,103 @@
+# Contributor Guidelines
+
+## Developer Setup
+
 The easiest way to work on Ultra is to clone this repo and navigate to the
-`/workspace` directory. This is simple setup of an Ultra project, and imports
-Ultra from a relative URL.
+`/workspace` directory. It contains a simple webapp that imports Ultra from a
+relative URL.
 
 - `cd workspace`
 - `deno task dev`
 
 ---
 
-Join the Discord if you would like to chat about your PR.
+## Development Standards
+
+### Pull Requests
+
+Before submitting a PR make sure that the following commands are run without
+error:
+
+```
+deno fmt
+deno lint
+deno task test
+deno task test:puppeteer
+```
+
+Note that `deno fmt` needs to be run on any changes to markdown or other
+non-source code files.
+
+The name or description of a Pull Request should be prefixed (pre-appended) with
+one of the following tokens (adapted from the
+[Conventional Commits standard](https://www.conventionalcommits.org/en/v1.0.0/)):
+
+1. **feat:** - PR is a new Ultra feature.
+2. **fix:** - PR is a bug fix.
+3. **doc:** - PR is a documentation change.
+4. **test:** - PR contains only unit and/or end-to-end tests.
+5. **chore:** - PR involves code cleanup or minor changes.
+6. **style:** - PR involves formatting or linting changes.
+7. **refactor:** - PR contains code refactoring.
+
+Please note:
+
+- The prefix token must be followed by a colon (:).
+- Add an exclamation point (!) before the colon if the PR contains a breaking
+  change.
+- An optional one-word scope indicator may be added in parentheses prior to the
+  colon (and the '!' if used).s
+
+Note that a PR might contain code that covers a number of overlapping areas,
+such as one involving a new feature and containing tests. Select the prefix that
+encompasses the majority of the PR's code.
+
+Although not required, it is a good idea to prefix commit descriptions too.
 
-Please ensure you run `deno lint` and `deno fmt` before opening any PRs.
+Here are some PR naming examples with a proper prefix ():
+
+```
+feat: Add CSS compression
+fix: End-to-end test fixes
+doc(contrib): Update contributor guidelines
+chore(build): Update Deno version
+refactor!: Convert Ultra to Angular
+```
+
+Naming PRs in this fashion will help in the creation of release notes.
+
+### Coding Style
+
+Our coding style standard follows the
+[Deno style guide](https://deno.land/manual/contributing/style_guide.md) (DSG).
+Please pay particular attention to the following style guide items:
+
+- [JSDoc comments](https://jsdoc.app/) on all exported functions. (see
```

Comment:
I recommend dropping the link to JSDoc and just using the one to Deno, and also explicitly stating that JSDoc-_style_ **descriptions** are required, but not the tags for arguments and return types, which TS covers.

Reply:
Yes, that's a good point.Frankly, I don't like the term JSDocs in the Deno Style Guide as these are pretty much just plain JS comments.